fendermint/vm/interpreter/src/fvm/state/genesis.rs

```diff
@@ -176,6 +176,74 @@ where
         }
     }
 
+    /// Replaces the built in actor with custom actor. This assumes the system actor is already
+    /// created, else it would throw an error.
+    pub fn replace_builtin_actor(
+        &mut self,
+        built_in_actor_name: &str,
+        built_in_actor_id: ActorID,
+        custom_actor_name: &str,
+        state: &impl Serialize,
+        balance: TokenAmount,
+        delegated_address: Option<Address>,
+    ) -> anyhow::Result<()> {
+        let code_cid = self
+            .update_system_actor_manifest(built_in_actor_name, custom_actor_name)
+            .context("failed to replace system actor manifest")?;
+
+        self.create_actor_internal(
+            code_cid,
+            built_in_actor_id,
+            state,
+            balance,
+            delegated_address,
+        )
+    }
+
+    /// Update the manifest id of the system actor, returns the code cid of the replacing
+    /// custom actor.
+    fn update_system_actor_manifest(
+        &mut self,
+        built_in_actor_name: &str,
+        custom_actor_name: &str,
+    ) -> anyhow::Result<Cid> {
+        let code = *self
+            .custom_actor_manifest
+            .code_by_name(custom_actor_name)
+            .ok_or_else(|| anyhow!("replacement {custom_actor_name} actor not found"))?;
+
+        let manifest_cid = self
+            .get_actor_state::<system::State>(system::SYSTEM_ACTOR_ID)?
+            .builtin_actors;
+
+        let mut built_in_actors: Vec<(String, Cid)> = self
+            .store()
+            .get_cbor(&manifest_cid)
+            .context("could not load built in actors")?
+            .ok_or_else(|| anyhow!("cannot find manifest cid {}", manifest_cid))?;
+
+        for (_, code_cid) in built_in_actors
+            .iter_mut()
+            .filter(|(n, _)| n == built_in_actor_name)
+        {
+            *code_cid = code
+        }
+
+        let builtin_actors = self.put_state(built_in_actors)?;
+        let new_cid = self.put_state(system::State { builtin_actors })?;
+        let mutate = |actor_state: &mut ActorState| {
+            actor_state.state = new_cid;
+            Ok(())
+        };
```

Comment:
Where do we apply this mutation?

Reply:
Here: https://github.com/consensus-shipyard/ipc/pull/676/files#diff-8f0901d357789002b95cef8fd68d18055c0ef0b06a3873ee9cea6396f177ea3cR240-R241

Reply:
Oops, my bad 😆